chalice/deploy/deployer.py

```diff
@@ -324,18 +325,19 @@ def _first_time_lambda_create(self, config, function_name, stage_name):
             config.project_dir)
         zip_contents = self._osutils.get_file_contents(
             zip_filename, binary=True)
-        return self._aws_client.create_function(
-            function_name, role_arn, zip_contents,
-            config.environment_variables,
-            config.lambda_python_version,
-            self._function_tags(config, stage_name),
-        )
-
-    def _function_tags(self, config, stage_name):
-        # type: (Config, str) -> Dict[str, str]
-        tag = 'version=%s:stage=%s:app=%s' % (chalice_version,
-                                              stage_name, config.app_name)
-        return {'aws-chalice': tag}
+        client_kwargs = {
```

Comment:
I'd prefer not to do that (setting defaults) at the config layer.  We've ran into this several times in the past (botocore config) where we can't differentiate between whether a value was explicitly set *or* the value is a default value. We need to have at least one layer somewhere that can differentiate between those two cases.

I would prefer moving the `if not None` checks into the `create_function` call.  We need that kwargs style anyways for botocore client calls, and it moves the if checks down a layer.

If you want to modify the `runtime` value to be consistent I think that's fine as well (there's no backwards compatibility issues with this API).